models.py

```diff
@@ -0,0 +1,96 @@
+""" Helper module to manage torch vision models """
+from typing import Optional
+
+import torch
+import torchvision.models as models
+import torch.nn as nn
+import numpy as np
+
+from torchvision.models.resnet import __all__ as all_resnet_models
+from torchvision.models.alexnet import __all__ as all_alexnet_models
+from torchvision.models.vgg import __all__ as all_vgg_models
+from torchvision.models.squeezenet import __all__ as all_squeezenet_models
+from torchvision.models.densenet import __all__ as all_densenet_models
+from torchvision.models.mnasnet import __all__ as all_mnasnet_models
+from torchvision.models.mobilenet import __all__ as all_mobilenet_models
+from torchvision.models.googlenet import __all__ as all_googlenet_models
+
+
+class EmbeddingModelWrapper:
+    """
+    The ``EmbeddingModelWrapper`` acts as an unified interface to the `torchvision` models.
+    It hides the model specific logic for computing embeddings from the user.
+
+    :param model_name: Name of the `torchvision` model. Classnames are not allowed, i.e.
+                       use `resnet_18` instead of `ResNet`.
+    :param device: Which device the model runs on. Can be 'cpu' or 'cuda'
+    """
+    def __init__(self, model_name: str, device: Optional[str] = None):
+        if not device:
+            device = 'cuda' if torch.cuda.is_available() else 'cpu'
+
+        self._layer_name = _ModelCatalogue.get_layer_name(model_name)
+        self._model = getattr(models, model_name)(pretrained=True)
+
+        self.device = device
+
+        self._pooling_layer = nn.AdaptiveAvgPool2d(1)
```

Comment:
try with  `nn.AdaptiveAvgPool2d(output_size=(1, 1))`

Reply:
Changed in https://github.com/jina-ai/executor-image-torch-encoder/pull/7/commits/e20c9c7b4853241f9516cf32c40b7e7888bc3885